### PR TITLE
[mac][mapkit] Allow creation of MKLocalSearchCompleter instance

### LINKED ISSUE #8274: macOS: cannot instantiate MKLocalSearchCompleter
### Steps to Reproduce
`var searchCompleter = new MKLocalSearchCompleter();
`
### Expected Behavior
Compiles (it does compile on iOS)

### Actual Behavior
Does not compile
Perhaps caused by this line: https://github.com/xamarin/xamarin-macios/blob/63d092d5268a53acc601f54f58b2db96c7e169f6/src/mapkit.cs#L1901
Is there no way to create a MKLocalSearchCompleter on macOS?

### PR DESCRIPTION
**Pull request by chamons**
- Fix #8274
- Turns out that MKLocalSearchCompleter can be instanced, even on iOS
	- Tested on sim and in docs
- MKLocalSearchCompletion is still not instanced, updating comment

See:
- https://developer.apple.com/documentation/mapkit/mklocalsearchcompleter?language=objc
- https://developer.apple.com/documentation/mapkit/mklocalsearchcompletion?language=objc

### CODE REVIEW

File: src/mapkit.cs [hunk 1]
```diff
@@ -1953,7 +1951,7 @@ interface MKLocalSearchCompleterDelegate {
 
 	[TV (9,2)][NoWatch][iOS (9,3)]
 	[BaseType (typeof(NSObject))]
-#if MONOMAC || XAMCORE_3_0 // Avoid breaking change in iOS
+#if MONOMAC || XAMCORE_3_0 // "You do not create instances of this class directly"
```
Comment (stephen-hawley) [inline src/mapkit.cs:1954]: <Boromir>"One does not simply create instances of this class directly"</Boromir>

Comment (stephen-hawley) [reply src/mapkit.cs:1954]: --Boromir

File: src/mapkit.cs [hunk 2]
```diff
@@ -1897,9 +1897,7 @@ partial interface MKTileOverlayRenderer {
 
 	[TV (9,2)][NoWatch][iOS (9,3)][Mac(10,11,4)]
 	[BaseType (typeof (NSObject))]
-#if MONOMAC || XAMCORE_3_0 // Avoid breaking change in iOS
 	[DisableDefaultCtor]
```
Comment (rolfbjarne) [inline src/mapkit.cs:1900]: You want to remove this line too:

```suggestion
```

**Comment by chamons** [comment]
The PR title is wrong, I had it written up before I understood what was going on. I have not tracked down the original commit down, i'll do that tomorrow.

**Comment by spouliot** [comment]
The extra precompiler checks were added here
```
%git show e729f9ca90a
commit e729f9ca90a95b1fa9cfe9b418ff0d949912107e
Author: Alex Soto <alex@alexsoto.me>
Date:   Thu Mar 24 11:44:06 2016 -0600

    [iOS] Fix some breaking changes in iOS

diff --git a/src/avfoundation.cs b/src/avfoundation.cs
index cfeb492095..ee9555dbe7 100644
--- a/src/avfoundation.cs
+++ b/src/avfoundation.cs
@@ -8953,7 +8953,9 @@ namespace XamCore.AVFoundation {
 
        [iOS (9,3), Mac (10,11,3)]
        [BaseType (typeof(AVPlayerItemMediaDataCollector))]
+#if MONOMAC || XAMCORE_3_0 // Avoid breaking change in iOS
        [DisableDefaultCtor]
+#endif
        interface AVPlayerItemMetadataCollector
        {
                [DesignatedInitializer]
diff --git a/src/mapkit.cs b/src/mapkit.cs
index 60768ff8b8..1dc29d4149 100644
--- a/src/mapkit.cs
+++ b/src/mapkit.cs
@@ -1729,7 +1729,9 @@ namespace XamCore.MapKit {
 
        [TV (9,2)][NoWatch][iOS (9,3)][Mac(10,11)]
        [BaseType (typeof (NSObject))]
+#if MONOMAC || XAMCORE_3_0 // Avoid breaking change in iOS
        [DisableDefaultCtor]
+#endif
        interface MKLocalSearchCompleter {
                [Export ("queryFragment")]
                string QueryFragment { get; set; }
@@ -1772,7 +1774,9 @@ namespace XamCore.MapKit {
 
        [TV (9,2)][NoWatch][iOS (9,3)]
        [BaseType (typeof(NSObject))]
+#if MONOMAC || XAMCORE_3_0 // Avoid breaking change in iOS
        [DisableDefaultCtor]
+#endif
        interface MKLocalSearchCompletion {
                [Export ("title", ArgumentSemantic.Strong)]
                string Title { get; }
diff --git a/src/storekit.cs b/src/storekit.cs
index e1dc464d37..545290060b 100644
--- a/src/storekit.cs
+++ b/src/storekit.cs
@@ -438,7 +438,9 @@ namespace XamCore.StoreKit {
 
        [iOS (9,3)]
        [BaseType (typeof (NSObject))]
+#if MONOMAC || XAMCORE_3_0 // Avoid breaking change in iOS
        [DisableDefaultCtor]
+#endif
        interface SKCloudServiceController {
                [Static]
                [Export ("authorizationStatus")]
```

and the `[DisableDefaultCtor] in 
```
$ git show 040fad15f47
commit 040fad15f47dde6500c0653723a2449601bea3c0
Author: Timothy Risi <timothy.risi@gmail.com>
Date:   Wed Mar 2 10:22:05 2016 -0900

    Fix dontlink-mac test failures

diff --git a/src/avfoundation.cs b/src/avfoundation.cs
index 76e4b86f64..cfeb492095 100644
--- a/src/avfoundation.cs
+++ b/src/avfoundation.cs
@@ -8953,6 +8953,7 @@ namespace XamCore.AVFoundation {
 
        [iOS (9,3), Mac (10,11,3)]
        [BaseType (typeof(AVPlayerItemMediaDataCollector))]
+       [DisableDefaultCtor]
        interface AVPlayerItemMetadataCollector
        {
                [DesignatedInitializer]
diff --git a/src/foundation.cs b/src/foundation.cs
index 9f83f6b76d..3429181c07 100644
--- a/src/foundation.cs
+++ b/src/foundation.cs
@@ -9199,6 +9199,7 @@ namespace XamCore.Foundation
                [return: NullAllowed]
                NSObject ReverseTransformedValue ([NullAllowed] NSObject value);
 
+#if !MONOMAC
                [iOS (9, 3)]
                [Notification]
                [Field ("NSUserDefaultsSizeLimitExceededNotification")]
@@ -9213,6 +9214,7 @@ namespace XamCore.Foundation
                [Notification]
                [Field ("NSUbiquitousUserDefaultsCompletedInitialSyncNotification")]
                NSString CompletedInitialSyncNotification { get; }
+#endif
 
                [Notification]
                [Field ("NSUserDefaultsDidChangeNotification")]
diff --git a/src/mapkit.cs b/src/mapkit.cs
index e520edf721..c5a9c77040 100644
--- a/src/mapkit.cs
+++ b/src/mapkit.cs
@@ -1671,6 +1671,7 @@ namespace XamCore.MapKit {
 
        [TV (9,2)][NoWatch][iOS (9,3)]
        [BaseType (typeof (NSObject))]
+       [DisableDefaultCtor]
        interface MKLocalSearchCompleter {
                [Export ("queryFragment")]
                string QueryFragment { get; set; }
@@ -1713,6 +1714,7 @@ namespace XamCore.MapKit {
 
        [TV (9,2)][NoWatch][iOS (9,3)]
        [BaseType (typeof(NSObject))]
+       [DisableDefaultCtor]
        interface MKLocalSearchCompletion {
                [Export ("title", ArgumentSemantic.Strong)]
                string Title { get; }
diff --git a/src/storekit.cs b/src/storekit.cs
index cf7ec27865..e1dc464d37 100644
--- a/src/storekit.cs
+++ b/src/storekit.cs
@@ -435,10 +435,10 @@ namespace XamCore.StoreKit {
                [Export ("productViewControllerDidFinish:"), EventArgs ("SKStoreProductViewController")]
                void Finished (SKStoreProductViewController controller);
        }
-#endif
 
        [iOS (9,3)]
        [BaseType (typeof (NSObject))]
+       [DisableDefaultCtor]
        interface SKCloudServiceController {
                [Static]
                [Export ("authorizationStatus")]
@@ -465,4 +465,5 @@ namespace XamCore.StoreKit {
                [Field ("SKCloudServiceCapabilitiesDidChangeNotification")]
                NSString CloudServiceCapabilitiesDidChangeNotification { get; }
        }
+#endif
 }
```

might be worth to check the other changes too

**Comment by spouliot** [comment]
@chamons you added the `[Mac(10,11,4)]` on that type in a later commit. 

This is (likely) why it failed (in XM tests) before and the original commit (to remove the default `.ctor`) was incorrect.

**Comment by chamons** [comment]
The commit in question (e729f9ca90a) had four changes, two of them touched by this PR and two more:

- AVPlayerItemMetadataCollector - The documentation shows it's use, so it's also wrong...
- SKCloudServiceController - Examples in swift suggest default constructor works as well, so also wrong.

Let me do a second PR for fixing those. Filed as https://github.com/xamarin/xamarin-macios/issues/8398

**Comment by chamons** [comment]
build

**Comment by chamons** [comment]
Errors were HE0038.

**Pull request merged**